src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const statesArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties': {
+        const currentState = i ? { ...statesArray[i - 1] } : { ...state };
+
+        Object.assign(currentState, actions[i].extraData);
+        statesArray.push(currentState);
+        break;
```

Comment:
1. no needs to update state at each case
2. no needs to push updates at each case
3. apply similar solution to cases below
```suggestion
        Object.assign(stateCopy, action.extraData);
        break;
```